client/js/components/procedure/admin/InstitutionTagManagement/TagListItem.vue

```diff
@@ -0,0 +1,313 @@
+      </template>
+    </div>
+  </form>
+</template>
+
+<script>
+import {
+  DpButton,
+  DpIcon,
+  DpInput,
+  dpValidateMixin
+} from '@demos-europe/demosplan-ui'
+import { mapActions, mapMutations, mapState } from 'vuex'
+
+export default {
+  name: 'TagListItem',
+
+  components: {
+    DpButton,
+    DpIcon,
+    DpInput
+  },
+
+  mixins: [dpValidateMixin],
+
+  props: {
+    /**
+     * If item is a category, it also has the property 'children'
+     * If item is a tag, it also has the properties 'categoryId' and 'isUsed'
+     */
+    item: {
+      type: Object,
+      required: true,
+      validator: (item) => {
+        return item.id && item.type && item.name
+      }
+    }
+  },
+
+  data () {
+    return {
+      isEditing: false,
+      name: this.item.name
+    }
+  },
+
+  computed: {
+    ...mapState('InstitutionTagCategory', {
+      institutionTagCategories: 'items'
+    }),
+
+    ...mapState('InstitutionTag', {
+      institutionTags: 'items'
+    })
+  },
+
+  methods: {
+    ...mapActions('InstitutionTagCategory', {
+      deleteInstitutionTagCategory: 'delete',
+      restoreTagCategoryFromInitial: 'restoreFromInitial',
+      saveInstitutionTagCategory: 'save'
+    }),
+
+    ...mapActions('InstitutionTag', {
+      deleteInstitutionTag: 'delete',
+      restoreTagFromInitial: 'restoreFromInitial',
+      saveInstitutionTag: 'save'
+    }),
+
+    ...mapMutations('InstitutionTagCategory', {
+      updateInstitutionTagCategory: 'setItem'
+    }),
+
+    ...mapMutations('InstitutionTag', {
+      updateInstitutionTag: 'setItem'
+    }),
+
+    abort () {
+      this.name = this.item.name
+      this.isEditing = false
+    },
+
+    confirmAndDeleteCategory () {
+      if (dpconfirm(Translator.trans('check.category.delete', { categoryTitle: this.item.name }))) {
+        this.deleteCategory()
+          .then(() => {
+            this.$emit('item:deleted', this.item)
+          })
+      }
+    },
+
+    confirmAndDeleteCategoryWithTags () {
+      const { children, name } = this.item
+      const tagsInUse = children
+        .filter(tag => tag.isUsed)
+        .map(tag => tag.name)
+
+      if (tagsInUse.length > 0) {
+        if (dpconfirm(Translator.trans('check.category_with_tags_in_use.delete', { category: name, count: tagsInUse.length, tags: tagsInUse.join(', ') }))) {
+          this.deleteCategoryAndTags()
+        }
+      } else if (tagsInUse.length === 0) {
+        if (dpconfirm(Translator.trans('check.category_with_tags_not_in_use.delete', { category: name }))) {
+          this.deleteCategoryAndTags()
+        }
+      }
+    },
+
+    confirmAndDeleteTag () {
+      const { isUsed, name } = this.item
+
+      if (isUsed) {
+        if (dpconfirm(Translator.trans('check.tag_is_used.delete'))) {
+          this.deleteTag()
+        }
+      } else if (!isUsed) {
+        if (dpconfirm(Translator.trans('check.tag.delete', { tag: name }))) {
+          this.deleteTag()
+        }
+      }
```

Comment:
nitpick: 
I think this could be simplified a little bit:
```suggestion
      if (isUsed && dpconfirm(Translator.trans('check.tag_is_used.delete'))) {
          this.deleteTag()
      } else if (!isUsed && dpconfirm(Translator.trans('check.tag.delete', { tag: name }))) {
          this.deleteTag()
      }
```

Reply:
Now that you made me think about it, what do you think about this?

```
  const message = isUsed 
    ? Translator.trans('check.tag_is_used.delete') 
    : Translator.trans('check.tag.delete', { tag: name })

  if (dpconfirm(message)) {
    this.deleteTag()
  }
```

Reply:
I find it very elegant.

Reply:
Done in 68b82d47586ba85f938d53e8112b9d2dd95383b4